### PR TITLE
Feature/activity-composer-vue: Allow Composer extensibility

### PR DESCRIPTION
**Pull request by aelbenney**

### CODE REVIEW

File: component/core/src/main/java/org/exoplatform/social/core/jpa/storage/RDBMSActivityStorageImpl.java [hunk 1]
```diff
@@ -613,7 +613,21 @@ private void storeFile(Identity streamOwner, ExoSocialActivity activity) throws
             LOG.warn("No file storage plugin for datasource with name '{}'", activityFile.getStorage());
           }
         }
-        activityFileStoragePlugin.storeAttachments(activity, streamOwner, files.toArray(new ActivityFile[files.size()]));
+        if (StringUtils.isEmpty(activityFile.getId()) && StringUtils.isNotEmpty(activityFile.getUploadId())) {
+          try {
+            activityFileStoragePlugin.storeAttachments(activity, streamOwner, activityFile);
+          } catch (Exception e) {
+            LOG.error("Error while adding attachment to activity " + activityFile.getId(), e);
```
Comment (thomasdelhomenie) [inline component/core/src/main/java/org/exoplatform/social/core/jpa/storage/RDBMSActivityStorageImpl.java]: activityFile.getId() is always empty in this case.

Comment (thomasdelhomenie) [inline component/core/src/main/java/org/exoplatform/social/core/jpa/storage/RDBMSActivityStorageImpl.java]: activityFile.getId() is always empty in this case.

File: component/core/src/main/java/org/exoplatform/social/core/jpa/storage/RDBMSActivityFileStoragePlugin.java [hunk 2]
```diff
@@ -70,4 +70,8 @@ public void storeAttachments(ExoSocialActivity activity, Identity streamOwner, A
     }
   }
 
+  public void attachExistingFiles(ExoSocialActivity activity, Identity streamOwner, ActivityFile... attachments) throws Exception {
+    throw new Exception("Not implemented yet");
```
Comment (thomasdelhomenie) [inline component/core/src/main/java/org/exoplatform/social/core/jpa/storage/RDBMSActivityFileStoragePlugin.java]:
```suggestion
    throw new UnsupportedOperationException("Not implemented yet");
```

File: component/core/src/main/java/org/exoplatform/social/core/storage/ActivityFileStoragePlugin.java [hunk 3]
```diff
@@ -55,6 +55,16 @@ public abstract void storeAttachments(ExoSocialActivity activity,
...
 
+  /**
+   * Attach existing files to a given activity
+   *
+   * @param activity
+   * @param streamOwner
+   * @param attachments
+   * @throws Exception
+   */
+  public abstract void attachExistingFiles(ExoSocialActivity activity, Identity streamOwner, ActivityFile... attachments) throws Exception;
```
Comment (thomasdelhomenie) [inline component/core/src/main/java/org/exoplatform/social/core/storage/ActivityFileStoragePlugin.java]: Why not using a list for the attachments argument ?

File: webapp/portlet/src/main/webapp/activity-composer-app/components/ExoActivityComposer.vue [hunk 4]
```diff
@@ -23,6 +23,24 @@
               <i class="uiIconError"></i>{{ $t('activity.composer.post.error') }}
             </div>
           </transition>
+          <div v-if="attachments.length" class="attachments-list">
```
Comment (thomasdelhomenie) [inline webapp/portlet/src/main/webapp/activity-composer-app/components/ExoActivityComposer.vue]:
```suggestion
          <div v-if="attachments.length" class="attachmentsList">
```

File: webapp/portlet/src/main/webapp/activity-composer-app/components/ExoActivityComposer.vue [hunk 5]
```diff
@@ -23,6 +23,24 @@
...
+            <i class="uiIconAttach"></i>
+            <p class="attachedFiles">{{ $t('attachments.drawer.title') }} ({{ attachments.length }})</p>
+          </div>
+          <div class="composerApps">
```
Comment (thomasdelhomenie) [inline webapp/portlet/src/main/webapp/activity-composer-app/components/ExoActivityComposer.vue]: I would call them `composerActions` instead ? It is not necessary an app. WDYT ?

File: webapp/portlet/src/main/webapp/activity-composer-app/components/ExoActivityComposer.vue [hunk 6]
```diff
@@ -23,6 +23,24 @@
...
+            <div v-for="app in activityComposerApplications" :key="app.key" :class="`${app.appClass}App`">
+              <div class="appsItem" @click="openApp(app)">
```
Comment (thomasdelhomenie) [inline webapp/portlet/src/main/webapp/activity-composer-app/components/ExoActivityComposer.vue]:
```suggestion
              <div class="appsItem" @click="executeAction(app)">
```

**Review by thomasdelhomenie — CHANGES_REQUESTED** [review]

File: component/core/src/main/java/org/exoplatform/social/core/jpa/storage/RDBMSActivityStorageImpl.java (hunk 1 above)
Comment (aelbenney) [reply component/core/src/main/java/org/exoplatform/social/core/jpa/storage/RDBMSActivityStorageImpl.java]: Changed to `getName()`

File: webapp/portlet/src/main/webapp/activity-composer-app/components/ExoActivityComposer.vue (hunk 5 above)
Comment (aelbenney) [reply webapp/portlet/src/main/webapp/activity-composer-app/components/ExoActivityComposer.vue]: Yes, it's more clear!

**Review by thomasdelhomenie — APPROVED** [review]

**Pull request merged**